packages/@glimmer/compiler-delegates/src/module-unification/compiler-delegate.ts

```diff
@@ -39,8 +39,12 @@ export default class MUCompilerDelegate implements AppCompilerDelegate<TemplateM
   }
 
   protected _builtins() {
+    let mainLocator = this.templateLocatorFor({
+      module: '@glimmer/application',
+      name: 'mainTemplate'
+    });
     return {
-      main: helperLocatorFor('main', 'mainTemplate'),
+      main: mainLocator,
```

Comment:
Why is the built-in called `main` but it's called `mainTemplate` everywhere else?

I think we should generally try to avoid special casing the main template and just use the built-ins infrastructure where it exists.